cdc/http_handler.go

```diff
@@ -15,29 +15,64 @@ package cdc
 
 import (
 	"net/http"
+	"strconv"
 
 	"github.com/coreos/etcd/clientv3/concurrency"
 	"github.com/pingcap/errors"
+	"github.com/pingcap/ticdc/cdc/model"
+)
+
+const (
+	opVarAdminJob     = "job"
+	opVarChangefeedID = "cf-id"
 )
 
 type commonResp struct {
 	Status  bool   `json:"status"`
 	Message string `json:"message"`
 }
 
+func handleOwnerResp(w http.ResponseWriter, err error) {
+	if err != nil {
+		if errors.Cause(err) == concurrency.ErrElectionNotLeader {
+			writeError(w, http.StatusBadRequest, err)
+			return
+		}
+		writeInternalServerError(w, err)
+		return
+	}
+	writeData(w, commonResp{Status: true})
+}
+
 func (s *Server) handleResignOwner(w http.ResponseWriter, req *http.Request) {
 	if req.Method != http.MethodPost {
 		writeError(w, http.StatusBadRequest, errors.New("this api only supports POST method"))
 		return
 	}
 	err := s.capture.ownerManager.ResignOwner(req.Context())
+	handleOwnerResp(w, err)
+}
+
+func (s *Server) handleChangefeedAdmin(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
```

Comment:
We can restrict HTTP methods by adding a call to `Methods("POST")` after `HandleFunc`.

Reply:
default http `serverMux` does not provide `methodMatcher`

Reply:
We can use `gorilla/mux` later.